### PR TITLE
Add `download_large_files` in `monai/bundle/scripts.py`

### LINKED ISSUE #6362: Bundle Download Large Files
**Is your feature request related to a problem? Please describe.**
Many bundles have the `large_files.yml` file specifying large files (specifically pre-trained weights). The bundles in the GenerativeModels repo are kept separately currently from those in the model zoo so the bundle download program doesn't work quite as expected so users would want to use bundles from a checkout of the repo. These bundles will be moved over but other users may want their own separate hosting for bundles so I feel there is still a need to make it easier to start using a bundle from a checkout. 

One step for this is to make it easier to download these large files on the command line or programmatically. This would help people working with checkout code and make it easier to get bundles and their data into Colab and other environments. 

I would also suggest changes to the `download` command we have for bundles to work with a URL pointing to a directory in an arbitrary repo. This is much more complicated now but would allow a user to provide [this](https://github.com/Project-MONAI/GenerativeModels/tree/main/model-zoo/models/brain_image_synthesis_latent_diffusion_model) as the URL and get that whole bundle large files and all. Again this would be helpful in Colab and elsewhere.

**Describe the solution you'd like**
I would like the code for downloading the large files [here](https://github.com/Project-MONAI/model-zoo/blob/131b0da3f540ed13a8a1c02d8395e92a682d47cf/ci/utils.py#L93) to be put into a command in the `monai/bundle/scripts.py` file so that users can run this and get the files they need to use the bundle.

**Describe alternatives you've considered**
One has to manually download weights and other files currently which isn't impossible but adds an extra layer of complexity to tell people to look into the yaml file. 

**Additional context**
This was from experience trying to get going quickly with a bundle. We do also need a way of programmatically getting a bundle's network with pre-trained weights like `torch.hub.load` though this will require some added specification in the bundle.

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #6362 .

### Description

Add `download_large_files` in `monai/bundle/scripts.py`.
Refer to the one in https://github.com/Project-MONAI/model-zoo/blob/131b0da3f540ed13a8a1c02d8395e92a682d47cf/ci/utils.py#L93.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/bundle/scripts.py [hunk 1]
```diff
@@ -1619,3 +1619,42 @@ def create_workflow(
     workflow_.initialize()
 
     return workflow_
+
+
+def download_large_files(bundle_path: str, large_file_name: str | None = None) -> None:
```
Comment (wyli) [inline monai/bundle/scripts.py:1624]: I think the bundle path could be 'os.getcwd()' if it's not specified?

File: monai/bundle/scripts.py [hunk 2]
```diff
@@ -1619,3 +1619,42 @@ def create_workflow(
...
+    Args:
+        bundle_path: The path to the bundle where the files are located.
+        large_file_name: (Optional) The name of the large file to be downloaded.
+
+    """
+    if large_file_name is None:
+        large_file_path = list(Path(bundle_path).glob("large_files*"))
+        large_file_path = list(filter(lambda x: x.suffix in [".yml", ".yaml", ".json"], large_file_path))
+        if len(large_file_path) == 0:
+            raise FileNotFoundError("Cannot find the large files.")
```
Comment (wyli) [inline monai/bundle/scripts.py:1647]: The error message could be more specific, display the folder searched and no large_files.yml/JSON found. The user may ignore the error if the files exist or add the missing large_files config

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

File: monai/bundle/scripts.py (hunk 2 above)
Comment (KumoLiu) [reply monai/bundle/scripts.py]: Thanks, have updated it in the latest commit.

File: monai/bundle/scripts.py (hunk 1 above)
Comment (KumoLiu) [reply monai/bundle/scripts.py]: Good point, have updated it in the latest commit. Thanks!

**Pull request merged**